### PR TITLE
chore: update actions/checkout to v2

### LINKED ISSUE #144: [Security] Workflow test.yaml is using vulnerable action actions/checkout
The workflow test.yaml is referencing action actions/checkout using references v1. However this reference is missing the commit [a6747255bd19d7a757dbdda8c654a9f84db19839](https://github.com/actions/checkout/commits/a6747255bd19d7a757dbdda8c654a9f84db19839) which may contain fix to the some vulnerability. 
The vulnerability fix that is missing by actions version could be related to: 
(1) CVE fix 
(2) upgrade of vulnerable dependency 
(3) fix to secret leak and others. 
Please consider to update the reference to the action.

### PR DESCRIPTION
**Pull request by dyladan**
Fixes #144

### CODE REVIEW

**Review by naseemkullah — APPROVED** [review]

**Comment by vmarchaud** [comment]
Is this ready to review since there are debug `ls` in the workflows ? I believe not but i'm not sure

**Comment by dyladan** [comment]
@vmarchaud I opened it as "ready" initially because it seemed so obvious, but i can't get the test to pass on browser or node 16. I'm getting weird permission errors even though the result of `id` shows that we are running as root.

**Comment by dyladan** [comment]
@vmarchaud it is now ready

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -36,11 +39,12 @@ jobs:
     container:
       image: circleci/node:12-browsers
     steps:
-      - name: Checkout
-        uses: actions/checkout@v1
       - name: Permission Setup
         run: sudo chmod -R 777 /github /__w
 
+      - name: Checkout
```
Comment (vmarchaud) [inline .github/workflows/test.yaml:45]: shouldn't this job requires to install git too ?

Comment (dyladan) [reply .github/workflows/test.yaml:48]: Actually this is just a leftover from the testing. I removed it.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -15,8 +15,14 @@ jobs:
     container:
       image: ${{ matrix.container }}
     steps:
+      - name: Permission Setup
+        run: chmod -R 777 /github /__w
```
Comment (legendecas) [inline .github/workflows/test.yaml:19]: Just out of curiosity: why do we need this?

**Review by legendecas — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 2 above)
Comment (dyladan) [reply .github/workflows/test.yaml:19]: Trust me I am also curious. I was having all sorts of problems with the checkout failing without it, but only on the node 16 container. Thats why there was so much thrash on this PR.

Comment (vmarchaud) [reply .github/workflows/test.yaml:19]: I'm not really confortable merging this, shoudn't this kind of issue be addressed upstream in the actions itself ?

Comment (dyladan) [reply .github/workflows/test.yaml]: I updated it to use setup-node instead of the official node docker container and that seems to have cleared it up.

**Review by vmarchaud — APPROVED** [review]

**Pull request merged**